### ISSUE TITLE
`extract-reads`: validate trimming parameters prior to extraction

### ISSUE THREAD

**Issue by nbokulich** [qiime2/q2-feature-classifier#119]
**Improvement Description**
If `min-length` > `trunc-len`, all reads will be filtered out. This should be caught at the outset, before any read extraction occurs.

**Current Behavior**
Extraction will run (slow), then trimming, then `min-length` filtering.

**Proposed Behavior**
Either raise an error at the outset if `min-length` > `trunc-len`, otherwise reconsider order of trimming and filtering parameters (should we extract -> filter -> trim instead?).

**Comment by Oddant1** [comment]
@nbokulich Assuming we go with raising the exception, I can deal with this issue and write a test for it quickly once #136 is merged. In the current master branch, the tests for this code are one large mess, one thing #136 does is disentangle them. The test parameters will need to be changed to accommodate this change because currently all tests will raise this exception. We can also spin off the test refactoring into a separate PR and merge that.

**Comment by nbokulich** [comment]
sounds good, we can wait for #136 , thanks!
